interface/super/edit_globals.php

```diff
@@ -7,175 +7,74 @@
  * @author    Rod Roark <rod@sunsetsystems.com>
  * @author    Brady Miller <brady.g.miller@gmail.com>
  * @author    Ranganath Pathak <pathak@scrs1.org>
+ * @author    Robert Down <robertdown@live.com>
  * @copyright Copyright (c) 2010 Rod Roark <rod@sunsetsystems.com>
  * @copyright Copyright (c) 2016-2019 Brady Miller <brady.g.miller@gmail.com>
  * @copyright Copyright (c) 2019 Ranganath Pathak <pathak@scrs1.org>
+ * @copyright Copyright (c) 2019 Robert Down <robertdown@live.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
 
-require_once("../globals.php");
-require_once("../../custom/code_types.inc.php");
-require_once("$srcdir/acl.inc");
-require_once("$srcdir/globals.inc.php");
-require_once("$srcdir/user.inc");
-require_once(dirname(__FILE__)."/../../myportal/soap_service/portal_connectivity.php");
+require_once "../globals.php";
+require_once "../../custom/code_types.inc.php";
+require_once "$srcdir/acl.inc";
+require_once "$srcdir/globals.inc.php";
+require_once "$srcdir/user.inc";
+require_once "../../myportal/soap_service/portal_connectivity.php";
 
 use OpenEMR\Common\Crypto\CryptoGen;
 use OpenEMR\Common\Csrf\CsrfUtils;
-use OpenEMR\Common\Logging\EventAuditLogger;
-use OpenEMR\Core\Header;
-use OpenEMR\OeUI\OemrUI;
 use Ramsey\Uuid\Uuid;
+use OpenEMR\Admin\Service\SettingsService;
+use Symfony\Component\HttpFoundation\Request;
+
+/** @var Request */
+$request = Request::createFromGlobals();
+
+// Grab Twig environment and add the admin templates dir with "admin" namespace
+$twig = $GLOBALS['twig'];
+$twig->getLoader()->addPath("../../src/Admin/templates/globals", "admin");
```

Comment:
Isn't src directory only supposed to contain classes. Wouldn't it make more sense for there to be a separate templates directory that is not in src? So maybe then `../../templates/admin/globals` ?

Reply:
No, `src` does not have to only contain only classes. Making a global templates directory is bad practice (making a global _anything_ directory is usually bad practice. The goal is to keep everything related to a module within itself. In this instance the `Admin` folder is a module, everything related to the Admin section should go there. By creating a monolithic templates directory we'd make it impossible to have a modular design

Reply:
hi @robertdown ,

We have composer set to seek PSR4 classes in src:
https://github.com/openemr/openemr/blob/master/composer.json#L98

Won't that cause potential issues by dropping hundreds of other non PSR4 class scripts in there regarding performance for class loading?

-brady

Reply:
No. Why would more classes lead to slower class loading? And where are you getting hundreds of non PSR4 classes from? Nothing in `templates` is a PHP class

Reply:
I simply mean that won't hundreds of files/scripts that are not classes get in the way of composer's seeking for classes?